### PR TITLE
Remove 'weave run', 'start' and 'restart'

### LINKED ISSUE #2351: weave run interprets container args
Arguments after the image name are meant for the container; they should not be interpreted by `weave run`.
To illustrate:

```
weave run -ti alpine /bin/sh --name=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
Container name too long to be used as hostname
45f745308a532a1b0c3c351741c64a64318860a107dc0e049789fe34fcfe215d
```

Unfortunately it is hard to know where the image name appears; it depends on exactly which `docker run` flags take a following parameter.

### LINKED ISSUE #2353: ditch 'weave run|start|restart'
I doubt very many users are using `weave run|start|restart`, and instead are using the proxy or plugin. The lack of "waiting for the interface" in `weave run` is an achilles heel (which is one reason we introduced the proxy), and it suffers from other issues such as #2351.

Is there anything `weave run|start|restart` can do that the proxy cannot?

### PR DESCRIPTION
**Pull request by bboreham**
Fixes #2353 
Fixes #2351 

We should announce this ahead of the release.

### CODE REVIEW

**Comment by bboreham** [comment]
I should put a 'this command has been removed' message in, to save confusion.

**Comment by rade** [comment]
`ipam_cidrs_or_die` is no longer used.

**Comment by bboreham** [comment]
Also docs need amending

**Comment by rade** [comment]
@abuehrle pls check there aren't any guides that reference these commands.

File: weave [hunk 1]
```diff
@@ -2009,8 +2009,12 @@ EOF
         NO_MULTICAST_ROUTE=
         collect_cidr_args "$@"
         shift $CIDR_ARG_COUNT
+        WITHOUT_DNS=
         while [ $# -gt 0 ]; do
             case "$1" in
+                --without-dns)
+                    WITHOUT_DNS=1
+                    ;;
```
Comment (rade) [inline weave]: Is this intentionally undocumented / left out from the usage?

Comment (bboreham) [inline weave]: fixed

File: test/280_with_or_without_dns_test.sh [hunk 2]
```diff
@@ -12,7 +12,9 @@ check_dns() {
     chk=$1
     shift
 
-    container=$(weave_on $HOST1 run $@ $IP/24 -t --add-host=$STATIC:$STATIC_IP $DNS_IMAGE /bin/sh)
+    container=$(docker_on $HOST1 run $(dns_args $HOST1 "$@") -dt $DNS_IMAGE /bin/sh)
+    weave_on $HOST1 attach $IP/24 "$@" --rewrite-hosts --add-host=$STATIC:$STATIC_IP $container
+    echo $chk $HOST1 $container $TARGET
```
Comment (marccarre) [inline test/280_with_or_without_dns_test.sh]: Given there is `$chk $HOST1 $container $TARGET` on line 18, is this:
a) just a temporary add which was accidentally checked in?
b) something we want to conditionally print, e.g. `[ -z "$DEBUG" ] || echo ...`? or
c) something we want to always print from now onwards?

Comment (bboreham) [inline test/280_with_or_without_dns_test.sh]: an accident

Comment (marccarre) [inline test/280_with_or_without_dns_test.sh]: Thanks for the confirmation. I'll remove it and then merge this PR.

**Pull request merged**